### ISSUE TITLE
Exact-match search for tags

### ISSUE THREAD

**Issue by emdash-ie** [LycheeOrg/Lychee#1369]
At the moment, clicking on a tag to view photos with that tag takes you to the search page, searching for that tag. However, this search is not an exact-match search for the tag. If another tag contains this tag as a substring, photos tagged with the other tag will also appear in the results.

For example: say I tag photos taken in summer with the tag `summer`, and photos taken in summer 2022 with `summer 2022`. Clicking the summer tag on one of the photos leads to a search that includes all photos tagged with `summer`, but also all tagged with `summer 2022`. I would like this to lead to an exact-match search instead, containing only photos tagged with `summer`.

**Comment by emdash-ie** [comment]
I have discovered this also affects tag albums.

**Comment by kamil4** [comment]
Yes, our handling of tags, as well as searching, is primitive and due for an overhaul.

My feeling is that search should be expanded to handle AND/OR-type keywords, and there also needs to be an ability to specify exact/non-exact matches as well as some form of pattern matching. Search queries currently match against all of title, description, and tags, and there should be an ability to restrict that. There should probably also be an ability to restrict search to just albums vs just photos.

Tags are currently completely unstructured, which makes support for them difficult to improve; the UI is not great either (one needs to type them in every time, creating an opportunity for typos). I expect that we will move to a system where each tag first needs to be defined and then one can add them to photos by selecting from the list of existing tags.

Tag albums could then support the same kind of expanded search queries as I wrote about above. Eventually tag albums could essentially turn into "saved searches", so to speak.

Anyway, these are just my random thoughts. Nobody is working on this at the moment, AFAIK.
